xdsl/ir/core.py

```diff
@@ -61,6 +61,12 @@ def attributes(self) -> Iterator[type[Attribute]]:
     def name(self) -> str:
         return self._name
 
+    @staticmethod
+    def split_name(name: str) -> tuple[str, str]:
+        names = name.split(".", 1)
+        assert len(names) == 2, f"Invalid operation or attribute name {name}"
+        return tuple(names)
```

Comment:
Oh no

Reply:
yeah that didn't go well

Reply:
```suggestion
        first, second = names
        return (first, second)
```

Reply:
any reason for this?

Reply:
Pyright isn't smart enough to see that when a list of length 2 is passed to the tuple constructor from an iterable, the resulting tuple will also have length 2 (`tuple[str, str]`) as opposed to dynamic (`tuple[str, ...]`).

Reply:
ah right, it fixes the pyright error